src/core/agent/modules/signify/signifyApi.ts

```diff
@@ -100,15 +104,19 @@ export class SignifyApi {
     };
   }
 
-  async createDelegationIdentifier(delegatorPrefix: string): Promise<string> {
-    const signifyName = utils.uuid();
-    const icpResult2 = await this.signifyClient
-      .identifiers()
-      .create(signifyName, { delpre: delegatorPrefix });
-    const operation = await icpResult2.op();
-    // no need to wait for operation to complete because it needs to be interacted from delegator
-    const delegatePrefix = operation.name.split(".")[1];
-    return delegatePrefix;
+  async createDelegationIdentifier(
+    delegatorPrefix: string
+  ): Promise<CreateIdentifierResult> {
+    try {
+      const signifyName = utils.uuid();
+      const operation = await this.signifyClient
+        .identifiers()
+        .create(signifyName, { delpre: delegatorPrefix });
+      await operation.op();
+      return { signifyName, identifier: operation.serder.ked.i };
+    } catch (err) {
+      throw new Error(SignifyApi.FAILED_TO_CREATE_DELEGATION_IDENTIFIER);
```

Comment:
What error can be thrown to lead to this? We shouldn't bury errors - this way we just have "Failed to create new delegation AID" but we don't know what the error - I would just let it throw